src/ecdh.rs

```diff
@@ -45,7 +45,7 @@ pub enum Algorithm {
     P521,
 }
 
-pub(crate) mod x25519 {
+pub mod x25519 {
```

Comment:
So these create a cyclic dependency.
Let's pull out libcrux-p256-ecdh etc. in here as well. I think that's enough to break the cycles?

Reply:
Should we pull out the `ecdh` module as a whole, or the individual curves separately?
`libcrux-kem` uses some items from `ecdh` such as `ecdh::p256_derive`, but these are just re-exports of the (previously crate private) curve sub-modules. I think it would perhaps make sense to pull out the `ecdh` module as a whole to avoid duplication of Error types etc and then to also re-export the additional necessary parts of the curve submodules there, private and public key types as far as I can see and `x25519::derive`. What do you think?

Reply:
Yeah, let's do only one `ecdh ` crate for now. We can split that up later if we really want to.

Reply:
Ah, trying to pull that out I've noticed that `ecdh` depends on the `kem` module itself since we have `TryFrom` conversions from KEM keys to ECDH keys for x25519, so just pulling out `ecdh` does not resolve the cyclic dependency. Do we need these conversions in that direction?

Reply:
That should either be in `kem` when it's `into`, or in `ecdh` when it's `from`. Both is ok, depends on the code. Propose an option that works and looks good? 🙂

Reply:
I've pulled out the `ecdh` module and made it a dependency for `libcrux`. The `libcrux-kem` is now independent of `libcrux`. I would also like to make it a dependency for `libcrux` (i.e. getting rid of the old `kem` module in libcrux) as a next step.

Reply:
I've replaced `libcrux`'s `kem` module with a re-export of the standalone crate now.